x-pack/plugins/apm/public/components/shared/agent_icon/get_serverless_icon.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import defaultIcon from '../span_icon/icons/default.svg';
+import lambdaIcon from './icons/lambda.svg';
+import azureFunctionsIcon from './icons/functions.svg';
+
+const serverlessIcons: { [key: string]: string } = {
```

Comment:
I'd probably create an `enum` here, something like:

```
export enum ServerlessProvider {
  AwsLambda: 'aws.lambda',
  AzureFunctions: 'azure.functions`
}
```

Then:
```suggestion
type ServerlessIcons = Record<ServerlessProvider, string>
const serverlessIcons: ServerlessIcons = {
```